plugins/vpc-eni/network/network_windows.go

```diff
@@ -198,6 +201,53 @@ func (nb *NetBuilder) FindOrCreateEndpoint(nw *Network, ep *Endpoint) error {
 		hnsEndpoint.PrefixLength = uint8(pl)
 	}
 
+	// Add ACL policies for blocking IMDS access through the endpoint.
+	if ep.BlockIMDS {
+		// Create an ACL policy to block traffic to instance metadata endpoint.
+		err = nb.addEndpointPolicy(hnsEndpoint,
+			hcsshim.ACLPolicy{
+				Type:            hcsshim.ACL,
+				Action:          hcsshim.Block,
+				Direction:       hcsshim.Out,
+				RemoteAddresses: instanceMetadataEndpoint,
+				Protocol:        256,
+				Priority:        200,
+			})
+		if err != nil {
+			log.Errorf("Failed to add endpoint ACL policy to block imds traffic: %v.", err)
+			return err
+		}
+
+		// Create an ACL policy to allow all incoming traffic.
+		err = nb.addEndpointPolicy(hnsEndpoint,
+			hcsshim.ACLPolicy{
+				Type:      hcsshim.ACL,
+				Action:    hcsshim.Allow,
+				Direction: hcsshim.In,
+				Protocol:  256,
```

Comment:
Does 256 mean all protocols? What's special about priority 300?
Let's define these magic numbers as constants in the beginning of this file.

Reply:
Yes 256 means all protocols as confirmed by Microsoft. 
We need to ensure that the priority of block rule is higher than the allow rules. The numbers 200 and 300 are as per the documentation provided by [Microsoft in this post](https://techcommunity.microsoft.com/t5/networking-blog/l2bridge-container-networking/ba-p/1180923).